app/ide-desktop/lib/dashboard/src/authentication/cognito.ts

```diff
@@ -411,9 +403,29 @@ function parseUserSession(
       email,
       clientId,
       expireAt,
+      refreshUrl,
       accessToken: session.getAccessToken().getJwtToken(),
       refreshToken: session.getRefreshToken().getToken(),
-      refreshUrl: new URL(domain).toString(),
+    }
+  }
+}
+
+/**
+ * Extract the refresh session endpoint URL from the JWT token payload
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-access-token.html
+ * @throws Error if the `iss` field of the payload is not a valid URL.
+ */
+function extractRefreshUrlFromSession(session: cognito.CognitoUserSession): string {
+  const { iss } = session.getAccessToken().payload
+
+  if (typeof iss !== 'string') {
+    throw new Error('Payload does not have an iss field.')
+  } else {
+    try {
+      new URL(iss)
```

Comment:
`return new URL(iss).href`

Reply:
btw. I just had a look at the linked doc and it says that the `iss` field has the form:
```
https://cognito-idp.<Region>.amazonaws.com/<your user pool ID>
```

Should our URL also include the pool ID or should that be stripped?

Reply:
Yea, on the second thought `return new URL(iss).toString()` looks even better

Reply:
Okay, I've just did a practical test - I rebased on top of your PR and checked the new `refresh_url` - it does respond with a successfully refreshed token.

So this seems OK.